src/main/java/com/example/demo/db/entities/AvailableShifts.java

```diff
@@ -79,13 +99,66 @@ public boolean isValid() {
     }
 
 
+    @Override
+    public boolean isValid(AvailableShifts toValidate) {
+        return toValidate.isValid();
+    }
+
+    @Override
+    public String toString() {
+        return "AvailableShifts{" +
+                "id=" + id +
+                ", weekNumber=" + weekNumber +
+                ", dayNumber=" + dayNumber +
+                ", startHour=" + startHour +
+                ", duration=" + duration +
+                ", employeeCount=" + employeeCount +
+                ", mangerCount=" + mangerCount +
+                '}';
+    }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
-        if (!(o instanceof AvailableShifts that)) return false;
-        return weekNumber.equals(that.weekNumber) && dayNumber.equals(that.dayNumber) && getStartHour().equals(that.getStartHour()) && getDuration().equals(that.getDuration());
+        if (o == null || getClass() != o.getClass()) return false;
+        AvailableShifts that = (AvailableShifts) o;
+        return Objects.equals(id, that.id) && Objects.equals(weekNumber, that.weekNumber) && Objects.equals(dayNumber, that.dayNumber) && Objects.equals(startHour, that.startHour) && Objects.equals(duration, that.duration) && Objects.equals(employeeCount, that.employeeCount) && Objects.equals(mangerCount, that.mangerCount);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, weekNumber, dayNumber, startHour, duration, employeeCount, mangerCount);
+    }
+
+    @Override
+    public int compareTo(@NotNull AvailableShifts o) {
```

Comment:
Use `@NonNull` (`spring.lang`) instead of `@NotNull`(`jakarta.constrains`)
 when you want `Java lang` null protection. they seem the same but they are not...